applications/MeshingApplication/custom_utilities/mmg_utilities.cpp

```diff
@@ -0,0 +1,2556 @@
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        rMMGMeshInfo.NumberOfTetrahedra = mMmgMesh->ne;
+        rMMGMeshInfo.NumberOfPrism = mMmgMesh->nprism;
+    } else { // Surfaces
+        rMMGMeshInfo.NumberOfTriangles = mMmgMesh->nt;
+    }
+
+    KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) << "\tNodes created: " << rMMGMeshInfo.NumberOfNodes << std::endl;
+    if (TMMGLibrary == MMGLibrary::MMG2D) { // 2D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    } else if (TMMGLibrary == MMGLibrary::MMG3D) { // 3D
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfTriangles + rMMGMeshInfo.NumberOfQuadrilaterals << "\n\tTriangles: " << rMMGMeshInfo.NumberOfTriangles << "\tQuadrilaterals: " << rMMGMeshInfo.NumberOfQuadrilaterals << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTetrahedra + rMMGMeshInfo.NumberOfPrism << "\n\tTetrahedron: " << rMMGMeshInfo.NumberOfTetrahedra << "\tPrisms: " << rMMGMeshInfo.NumberOfPrism << std::endl;
+    } else { // Surfaces
+        KRATOS_INFO_IF("MmgUtilities", EchoLevel > 0) <<
+        "Conditions created: " << rMMGMeshInfo.NumberOfLines << "\n" <<
+        "Elements created: " << rMMGMeshInfo.NumberOfTriangles << std::endl;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<MMGLibrary TMMGLibrary>
+IndexVectorType MmgUtilities<TMMGLibrary>::CheckNodes(
+    ModelPart& rModelPart,
+    const IndexType EchoLevel
+    )
+{
+    DoubleVectorMapType node_map;
+
+    IndexVectorType nodes_to_remove_ids;
+
+    DoubleVectorType coords(Dimension);
+
+    auto& r_nodes_array = rModelPart.Nodes();
+    const auto it_node_begin = r_nodes_array.begin();
+
+    for(SizeType i = 0; i < r_nodes_array.size(); ++i) {
+        auto it_node = it_node_begin + i;
+
+        const array_1d<double, 3>& r_coordinates = it_node->Coordinates();
+
+        for(IndexType i_coord = 0; i_coord < Dimension; i_coord++)
+            coords[i_coord] = r_coordinates[i_coord];
+
+        node_map[coords] += 1;
+
+        if (node_map[coords] > 1) {
+            nodes_to_remove_ids.push_back(it_node->Id());
+            KRATOS_WARNING_IF("MmgUtilities", EchoLevel > 0) << "The mode " << it_node->Id() <<  " is repeated"<< std::endl;
+        }
+    }
+
+    return nodes_to_remove_ids;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+IndexVectorType MmgUtilities<MMGLibrary::MMG2D>::CheckFirstTypeConditions(const IndexType EchoLevel)
+{
+    IndexVectorMapType edge_map;
```

Comment:
Does `EchoLevel` need to be a parameter? It isn't used here and in other places it seems to clash with the function name.

Reply:
I can remove it making it a member variable

Reply:
I think removing it would be the simplest solution, but I wouldn't make it a member. Your decision.